src/Disks/getOrCreateDiskFromAST.cpp

```diff
@@ -26,12 +26,20 @@ namespace
 {
     std::string getOrCreateDiskFromDiskAST(const ASTFunction & function, ContextPtr context)
     {
-        /// We need a unique name for a created custom disk, but it needs to be the same
-        /// after table is reattached or server is restarted, so take a hash of the disk
-        /// configuration serialized ast as a disk name suffix.
-        auto disk_setting_string = serializeAST(function, true);
-        auto disk_name = DiskSelector::TMP_INTERNAL_DISK_PREFIX
-            + toString(sipHash128(disk_setting_string.data(), disk_setting_string.size()));
+        std::string disk_name;
+        if (function.name == "disk")
+        {
+            /// We need a unique name for a created custom disk, but it needs to be the same
+            /// after table is reattached or server is restarted, so take a hash of the disk
+            /// configuration serialized ast as a disk name suffix.
+            auto disk_setting_string = serializeAST(function, true);
+            disk_name = DiskSelector::TMP_INTERNAL_DISK_PREFIX
+                + toString(sipHash128(disk_setting_string.data(), disk_setting_string.size()));
+        }
+        else
+        {
+            disk_name = function.name.substr(std::strlen("disk_"));
```

Comment:
```suggestion
            static constexpr std::string_view custom_disk_prefix = "disk_"; 
            disk_name = function.name.substr(custom_disk_prefix.size());
```
`size()` should be cheaper then searching for `\0`